runtime/lua/vim/lsp.lua

```diff
@@ -1309,50 +1315,61 @@ end
 
 nvim_command("autocmd VimLeavePre * lua vim.lsp._vim_exit_handler()")
 
+--- Get a list of all of the clients of a buffer that support the given method.
+---
+---@param bufnr (number) Buffer handle, or 0 for current.
+---@param method (string) LSP method name
+---@returns list of all client IDs attached to `bufnr` supporting `method`
+function lsp.buf_supported_clients(bufnr, method)
+  local supported_clients = {}
+  for_each_buffer_client(bufnr, function(client, client_id)
+    if client.supports_method(method) then
+      table.insert(supported_clients, client_id)
+    end
+  end)
+
+  return supported_clients
+end
 
 --- Sends an async request for all active clients attached to the
---- buffer.
+--- buffer that support the given method with client-specific params and handlers.
 ---
 ---@param bufnr (number) Buffer handle, or 0 for current.
 ---@param method (string) LSP method name
----@param params (optional, table) Parameters to send to the server
----@param handler (optional, function) See |lsp-handler|
+---@param params (optional, table client_id -> table) Map from client IDs to parameters
+---to send to the server for that client
+---@param handler (optional, table client_id -> function) Map from client IDs to handlers, see |lsp-handler|
 --  If nil, follows resolution strategy defined in |lsp-handler-configuration|
 --
 ---@returns 2-tuple:
 ---  - Map of client-id:request-id pairs for all successful requests.
 ---  - Function which can be used to cancel all the requests. You could instead
 ---    iterate all clients and call their `cancel_request()` methods.
-function lsp.buf_request(bufnr, method, params, handler)
+function lsp.buf_request_general(bufnr, method, params, handler)
```

Comment:
I don't like general, maybe `buf_request_mapping`? WDYT @mfussenegger

Reply:
If it weren't for the breaking change I'd say rename `buf_request` to `buf_request_unsafe` and use the old name for the new signature.

Reply:
why unsafe?